static/app/views/routeNotFound.tsx

```diff
@@ -1,44 +1,46 @@
-import {Component} from 'react';
+import {useEffect} from 'react';
 import DocumentTitle from 'react-document-title';
+import {RouteComponentProps} from 'react-router';
 import * as Sentry from '@sentry/react';
-import {Location} from 'history';
 
 import NotFound from 'app/components/errors/notFound';
 import Footer from 'app/components/footer';
 import Sidebar from 'app/components/sidebar';
+import {t} from 'app/locale';
 
-type Props = {
-  location: Location;
-};
+type Props = RouteComponentProps<{}, {}>;
+
+function RouteNotFound({router, location}: Props) {
+  useEffect(() => {
+    // Attempt to fix trailing slashes first
+    const {pathname, search, hash} = location;
+
+    if (pathname[pathname.length - 1] !== '/') {
+      router.replace(`${pathname}/${search}${hash}`);
+      return;
+    }
 
-class RouteNotFound extends Component<Props> {
-  componentDidMount() {
     Sentry.withScope(scope => {
```

Comment:
Will this run twice now? Once on mount and then again when the pathname is replaced?

Reply:
It doesn't since we early return after the router.replace, so we only fire this when there's already a trailing slash but it's a 404